core/templates/domain/objects/UnitsObjectFactory.ts

```diff
@@ -74,8 +74,13 @@ export class UnitsObjectFactory {
     for (var i = 0; i < units.length; i++) {
       if ('*/()# '.includes(units[i]) && unit !== 'per') {
         if (unit.length > 0) {
-          if ((unitList.length > 0) && this.isunit(unitList.slice(-1).pop())) {
-            unitList.push('*');
+          if (unitList.length > 0) {
+            var elt = unitList.slice(-1).pop();
+            if (typeof elt === 'string') {
+              if (this.isunit(elt)) {
+                unitList.push('*');
+              }
+            }
```

Comment:
Why this needs to be changed?

Reply:
Running the strict type checks with the original code gave the following error:

core/templates/domain/objects/UnitsObjectFactory.ts(77,52): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.

Type 'undefined' is not assignable to type 'string'.